tasks/SaveWrapper.ts

```diff
@@ -1,80 +1,116 @@
+            feederPoolAddresses,
+            fAssetAddresses,
+            mAssetToken.savings,
+            mAssetToken.vault,
         )
+        await logTxDetails(tx, "Approve mAsset")
+    })
+
+task("SaveWrapper.approveMulti", "Sets approvals for multiple tokens/a single spender")
+    .addParam(
+        "tokens",
+        "Comma separated symbols of the tokens that is being approved. eg USDC,DAI,USDT,sUSD",
+        undefined,
+        types.string,
+        false,
+    )
+    .addParam(
+        "spender",
+        "Token symbol of the mAsset or address type. eg mUSD, mBTC, feederPool, savings or vault",
+        undefined,
+        types.string,
+        false,
+    )
+    .addOptionalParam("speed", "Defender Relayer speed param: 'safeLow' | 'average' | 'fast' | 'fastest'", "fast", types.string)
+    .setAction(async (taskArgs, hre) => {
+        const chain = getChain(hre)
+        const signer = await getSigner(hre, taskArgs.speed)
+
+        const wrapperAddress = resolveAddress("SaveWrapper", chain)
+        const wrapper = SaveWrapper__factory.connect(wrapperAddress, signer)
+
+        const tokenSymbols = taskArgs.tokens.split(",")
+        const tokenAddresses = tokenSymbols.map((symbol) => resolveAddress(symbol, chain))
+
+        const spenderAddress = ["feederPool", "savings", "vault"].includes(taskArgs.spender)
+            ? resolveAddress(taskArgs.token, chain, taskArgs.spender) // token is mUSD or mBTC
+            : resolveAddress(taskArgs.spender, chain) // spender is mUSD or mBTC
+
+        const tx = await wrapper["approve(address[],address)"](tokenAddresses, spenderAddress)
+        await logTxDetails(tx, "Approve multiple tokens/single spender")
     })
 
 task("SaveWrapper.approve", "Sets approvals for a single token/spender")
-    .addParam("saveWrapper", "SaveWrapper address", undefined, params.address, false)
-    .addParam("token", "Token address", undefined, params.address, false)
-    .addParam("spender", "Spender address", undefined, params.address, false)
-    .setAction(async ({ saveWrapper, token, spender }: { saveWrapper: string; token: string; spender: string }, hre) => {
-        const signer = await getSigner(hre)
-        await sendTx(
-            SaveWrapper__factory.connect(saveWrapper, signer),
-            "approve(address,address)",
-            "Approve single token/spender",
-            token,
-            spender,
-        )
+    .addParam("token", "Symbol of the token that is being approved. eg USDC, WBTC, FEI, HBTC, mUSD, imUSD", undefined, types.string, false)
+    .addParam(
+        "spender",
+        "Token symbol of the mAsset or address type. eg mUSD, mBTC, feederPool, savings or vault",
+        undefined,
+        types.string,
+        false,
+    )
+    .addOptionalParam("speed", "Defender Relayer speed param: 'safeLow' | 'average' | 'fast' | 'fastest'", "fast", types.string)
+    .setAction(async (taskArgs, hre) => {
+        if (!taskArgs.spender) {
```

Comment:
it is good to add this kind of checks for future references , as it is easy to forget what should be passed or not